test/e2e/settings/no_auth_locations.go

```diff
@@ -51,16 +52,24 @@ var _ = framework.IngressNginxDescribe("No Auth locations", func() {
 		Expect(s).NotTo(BeNil())
 		Expect(s.ObjectMeta).NotTo(BeNil())
 
-		updateConfigmap(setting, noAuthPath, f.KubeClientSet)
-
 		bi := buildBasicAuthIngressWithSecondPath(host, f.Namespace.Name, s.Name, noAuthPath)
 		ing, err := f.EnsureIngress(bi)
 		Expect(err).NotTo(HaveOccurred())
 		Expect(ing).NotTo(BeNil())
+
+		if &defaultNginxConfigMapData == nil {
```

Comment:
Won't this always be true? I don't know if the if statement is necessary in all of these.

Reply:
Thought a bit more about this - I think it'd most likely always be false. The address of that stack variable isn't likely to be nil, since it's created as an empty map.

I think what I was thinking there was that the address would be nil until we assign a value to it - meaning it would only occur the first time - however I realize `defaultNginxConfigMapData` isn't a pointer.

Going to instead explicitly initialize the map to `nil` - should achieve what I was trying to without overcomplicating things :) Thanks Zenara!